core/templates/pages/library-page/search-results/activity-tiles-infinity-grid.component.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { of } from 'rxjs';
+import { WindowDimensionsService } from
+  'services/contextual/window-dimensions.service';
+
+describe('Activity tiles infinity grid component', function() {
+  var ctrl = null;
+  var $q = null;
+  var $rootScope = null;
+  var $scope = null;
+  var searchService = null;
+  var userService = null;
+  var windowDimensionsService = null;
+
+  var mockWindow = {
+    location: ''
+  };
+  var loadingMessageChangeEventEmitter = new EventEmitter();
+
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    $provide.value('$window', mockWindow);
+    $provide.value('LoaderService', {
+      onLoadingMessageChange: loadingMessageChangeEventEmitter
+    });
+  }));
+
+  beforeEach(function() {
+    windowDimensionsService = TestBed.get(WindowDimensionsService);
+  });
+
+  afterEach(function() {
+    ctrl.$onDestroy();
+  });
+
+  beforeEach(angular.mock.inject(function($injector, $componentController) {
+    $q = $injector.get('$q');
+    $rootScope = $injector.get('$rootScope');
+    searchService = $injector.get('SearchService');
+    userService = $injector.get('UserService');
+
+    spyOn(userService, 'getUserInfoAsync').and.returnValue($q.resolve({
+      isLoggedIn: () => true
+    }));
+
+    spyOn(windowDimensionsService, 'getResizeEvent').and.returnValue(
+      of(new Event('resize')));
+
+    $scope = $rootScope.$new();
+    ctrl = $componentController('activityTilesInfinityGrid', {
+      $scope: $scope,
+      WindowDimensionsService: windowDimensionsService
+    });
+  }));
+
+  it('should initialize correctly controller properties after its' +
+    ' initialization', function() {
+    spyOn(windowDimensionsService, 'getWidth').and.returnValue(470);
+    ctrl.$onInit();
+    expect(ctrl.libraryWindowIsNarrow).toBe(true);
+  });
+
+  it('should change container view when resizing page to more than 530 pixels',
+    function() {
+      spyOn(windowDimensionsService, 'getWidth').and.returnValue(768);
+      ctrl.$onInit();
+      angular.element(window).triggerHandler('resize');
+      expect(ctrl.libraryWindowIsNarrow).toBe(false);
+    });
+
+  it('should get more data from backend when reaching end of page and there' +
+    ' is more data to be loaded', function() {
+    ctrl.$onInit();
+    $rootScope.$broadcast('initialSearchResultsLoaded', [{}]);
+
+    var loadMoreDataSpy = spyOn(searchService, 'loadMoreData');
+    loadMoreDataSpy.and.callFake(
+      (successCallback, failCallback) => successCallback({
+        activity_list: [{}, {}, {}]
+      }, true));
+    ctrl.showMoreActivities();
+    $scope.$apply();
+
+    expect(ctrl.allActivitiesInOrder.length).toBe(4);
+    expect(ctrl.endOfPageIsReached).toBe(true);
+
+    loadMoreDataSpy.calls.reset();
+    loadMoreDataSpy.and.callThrough();
+    ctrl.showMoreActivities();
+    expect(searchService.loadMoreData).not.toHaveBeenCalled();
+  });
+
+  it('should get more data from backend after waiting page to be loaded' +
```

Comment:
I don't understand what "after waiting page to be loaded completely" means.. Can you clarify

Reply:
I changed the spec name to a better one.